casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -199,29 +196,41 @@ object InterpreterUtil {
       parents: Seq[BlockMessage],
       dag: BlockDagRepresentation[F],
       runtimeManager: RuntimeManager,
-      time: Option[Long],
       initStateHash: StateHash
   ): F[Either[Throwable, StateHash]] =
     for {
       blockHashesToApply <- findMultiParentsBlockHashesForReplay(parents, dag)
       blocksToApply      <- blockHashesToApply.traverse(b => ProtoUtil.unsafeGetBlock[F](b.blockHash))
-      deploys            = blocksToApply.flatMap(_.getBody.deploys.flatMap(ProcessedDeployUtil.toInternal))
-      replayResult <- ToAbstractContext[F].fromTask(
```

Comment:
what will happen when the `blocksToApply` collection holds more than 1 block?

Reply:
It will run `replayComputeState` as many times as there are `blocksToApply`.

Reply:
I believe that using the replay mechanic this way will result in further hits to performance. Essentially this makes the algorithm n^2 and forces the tuplespace to perform very expensive operations multiple times.
Is this the only way to solve passing time to replay?

Reply:
What is the expensive operation? `createCheckpoint`? `reset` in theory should be free.

Reply:
In the future we could have an optimization that compresses it back together if there is no deploy that uses the blockTime

Reply:
`reset` isn't free, because it has to re-populate previously cleared hot store with data

Reply:
Why is the hot store previously cleared? In 99% of cases, you're reseting to the previous checkpoint value and so if you just left the hot store alone, things would smoothly just continue.